datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -422,8 +422,10 @@ private void checkpoint(int partition, String checkpoint) {
   /**
    * Shuts down the event producer by flushing the checkpoints and closing the transport provider
    */
-  public void shutdown() {
-    _checkpointProvider.flush();
+  public void shutdown(boolean skipCheckpoint) {
+    if (!skipCheckpoint) {
+      _checkpointProvider.flush();
+    }
```

Comment:
You may want to add a comment on why we may want to skip checkpoint commit. (though this only affects if using the framework level checkpointing, which we don't use internally)

Feel free to update the comment where you pass 'skipCheckpoint' as 'true' instead of here if it feel like a better place to discuss this aspect.